design-guidelines/form-ux.md

```diff
@@ -0,0 +1,189 @@
+
+![Input field diagram](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram.png)
+
+### Label
+
+Labels are required to identify a corresponding input field. Label text should be descriptive, but as short as possible.
+
+A label must be provided to maintain screenreader accessibility, but it may be visually hidden if there are other cues that identify what the field does. For example: a text input with a magnifying glass and placeholder text that says "Search".
+
+![Input field diagram - label](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-label.png)
+
+Choice inputs such as checkboxes and segmented controls have a label for each option, but are also labeled as a collection.
+
+![Input field diagram - labels and legend](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-legend.png)
+
+### Required field indicator
+
+When there is a collection of optional and required fields, the required fields should be clearly marked. This can help the user avoid submitting an invalid form because they left a required field blank.
+
+![Input field diagram - required indicator](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-requiredIndicator.png)
+
+### Label tooltip
+
+Describes the purpose of the field and how it's value is used to help the user complete their task. This information will be hidden until the user opens the Tooltip, so do not put any information in here that is critical to task completion.
+
+![Input field diagram - tooltip](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-tooltip.png)
+
+### Input
+
+The input is the component that users interact with to set the value of the field.
+
+![Input field diagram - input](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-input.png)
+
+### Placeholder text
+
+When an input does not have a value set, placeholder text may be shown in it's place. Placeholder text is useful for hinting what data the input is expecting to receive. For example: showing "1" in the placeholder text for a field that is expecting a number.
+
+Labels may not be replaced by placeholder text unless there are enough visual cues or context clues for the user to understand what the input does. Refer to the "Label" section above for information on labeling inputs.
+
+![Input field diagram - placeholder](https://raw.githubusercontent.com/dcos-labs/ui-kit/master/design-guidelines/img/form-ux-patterns/inputFieldDiagram-placeholder.png)
+
+### Hint content
+
+Hint text can be used to caption an input with information that helps the user fill in the correct data. Hint text can also be used to display validation errors specific to the field.
```

Comment:
One or the other, but not both? Or are both allowed?

Reply:
Both are allowed. I'll add more specific guidance here.